x-pack/elastic-agent/pkg/composable/providers/docker/docker.go

```diff
@@ -0,0 +1,157 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package docker
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/bus"
+	"github.com/elastic/beats/v7/libbeat/common/docker"
+	"github.com/elastic/beats/v7/libbeat/common/safemapstr"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/errors"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/composable"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
+)
+
+func init() {
+	composable.Providers.AddDynamicProvider("docker", DynamicProviderBuilder)
+}
+
+type dockerContainerData struct {
+	container  *docker.Container
+	mapping    map[string]interface{}
+	processors []map[string]interface{}
+}
+type dynamicProvider struct {
+	logger *logger.Logger
+	config *Config
+}
+
+// Run runs the environment context provider.
+func (c *dynamicProvider) Run(comm composable.DynamicProviderComm) error {
+	watcher, err := docker.NewWatcher(c.logger, c.config.Host, c.config.TLS, false)
+	if err != nil {
+		// warn only; return nil (do nothing)
+		c.logger.Warnf("Failed starting docker provider: %s", err)
+		return nil
+	}
+	startListener := watcher.ListenStart()
+	stopListener := watcher.ListenStop()
+	stoppers := map[string]*time.Timer{}
+	stopTrigger := make(chan *dockerContainerData)
+
+	if err := watcher.Start(); err != nil {
+		// warn only; return nil (do nothing)
+		c.logger.Warnf("Failed starting docker provider: %s", err)
```

Comment:
On the Beats code based we tried no to use `Warn` as it is this strange middleground between info and error. Is user interaction needed or not? Is it a bug or not? I would suggest to move this to the Info level and phrase it more something like: Docker provider not started as docker not available (if we can make this distinction).

Reply:
Done!